lib/tasks/uploads.rake

```diff
@@ -907,6 +907,28 @@ task "uploads:recover" => :environment do
   end
 end
 
+task "uploads:ensure_correct_acl" => :environment do
```

Comment:
This rake task is my only real concern here as it could take a veeeery long time for people with lots and lots of uploads. I think each of these grouped calls should probably be in a transaction as well:

```
Upload.transaction do
  upload.update_secure_status
  upload.posts.each { |post| post.rebake! }
end
```

To speed it up I guess the uploads could be grouped and the code to update the status + rebake the post could be split into separate threads per group? Another nicety could be a counter of how many uploads there are and what progress we are up to, as it's hard to tell how far along this task is ATM. So far I have no real idea of the scale of data of real-world Discourse instances so I'm not sure if this is a valid worry.

Would be the intention here be to run this rake task manually on a server after the user has changed the secure media setting? If so should we add a message near the secure media setting to let them know that?

Reply:
Thanks @martin-brennan, I think you're right to raise a performance concern here. 

The intention is for this to be run manually, either when a relevant site setting has been changed (secure media, login required, etc.) or if a site has been moved, restored, and so on. Making the task faster and more verbose to the user running it (i.e. adding a progress status indicator) would be awesome!

Regarding the message near the secure media setting, I would wait and see if users raise this as an issue on meta often enough to warrant adding a note about the task in the UI.